Desktop/TypescriptExercises/day2 - angular form exercise/src/app/models/app.logic.ts

```diff
@@ -21,4 +21,42 @@ export class Logic {
       this.products.push(prd);
       return this.products;
     }
-}
+    searchProducts(searchText: string): Array<Product>
+    {
+      let searchResults: Array<Product> = []
+      this.products.forEach(element => {
+        if(element.Category.toLowerCase().includes(searchText.toLowerCase()) || element.ProductName.toLowerCase().includes(searchText.toLowerCase()))
+          {
+            searchResults.push(element)
+          }
+      });
+      return searchResults
+    }
+    fetchIndex(prd:Product): number
+    {
+      return this.products.map(function(e) { return e.ProductId }).indexOf(prd.ProductId);
```

Comment:
instead of map, you can use findIndexOf(predicate)

Reply:
done